examples/blockdiagperm.cc

```diff
@@ -0,0 +1,140 @@
+# {
+#   O1(b, k, nbyk1) +=! I(b, k, r_nbyk0) * W(k, r_nbyk0, nbyk1)
+#   O2(b, k, nbyk)   =  O1(b, Idxr(k, nbyk), Idxc(k, nbyk))
+# }
+
+# This TC requires LHS indexing which is a WIP + extra information that all
+# accesses are parallel (i.e. (IdxR, IdxC) form a permutation)
+# def blockdiagperm2dfused3(float(B, K, NBYK) I, float(K, NBYK, NBYK) W, float(K, NBYK) IdxR, float(K, NBYK) IdxC)  -> (O1)
+# {
+#  O1(b, IdxR(k, nbyk0), IdxC(k, nbyk0)) +=! I(b, k, r_nbyk0) * W(k, r_nbyk0, nbyk1)
+# }
+
+# This is the poor man's way of making things work today with a reshape
+# operation in between (in framework land).
+def blockdiagperm2dfissioned_1(float(B, K, NBYK) I, float(K, NBYK, NBYK) W)  -> (O)
+{
+  O(b, k, nbyk1) +=! I(b, k, r_nbyk0) * W(k, r_nbyk0, nbyk1)
+}
+
+def blockdiagperm2dfissioned_2(float(B, N) I, int32(N) Idx)  -> (O)
+{
+  O(b, n) = I(b, Idx(n)) where n in 0:N
+}
+
+  )TC";
+  tc::ATenCompilationUnit<tc::CudaTcExecutor> atCompl;
+  atCompl.define(tc);
+
+  // 1. Allocate and autotune
+  at::Tensor I = at::CUDA(at::kFloat).rand({128, 10, 50});
+  at::Tensor W = at::CUDA(at::kFloat).rand({10, 50, 50});
+  auto options = tc::CudaMappingOptions::makeNaiveCudaMappingOptions();
+  tc::autotune::GeneticAutotunerATen geneticAutotuneATen(tc);
+  auto bestOption = geneticAutotuneATen.tune(
+      "/tmp/save_results_blockdiagperm2dfissioned",
+      "blockdiagperm2dfissioned_1",
+      {I, W},
+      options);
+  auto handle = atCompl.compile(
+      "blockdiagperm2dfissioned_1", {I, W}, bestOption.getValue());
+  std::vector<at::Tensor> outputs;
+  auto duration =
+      atCompl.run("blockdiagperm2dfissioned_1", {I, W}, outputs, handle, true);
+
+  // 2. Allocate and autotune
+  at::Tensor O = outputs[0].clone().resize_({128, 500});
+  at::Tensor Idx = at::CPU(at::kInt).randperm({500}).toBackend(at::kCUDA);
+  tc::autotune::GeneticAutotunerATen geneticAutotuneATen2(tc);
+  auto bestOption2 = geneticAutotuneATen.tune(
+      "/tmp/save_results_blockdiagperm2dfissioned",
+      "blockdiagperm2dfissioned_2",
+      {O, Idx},
+      options);
+  auto handle2 = atCompl.compile(
+      "blockdiagperm2dfissioned_2", {O, Idx}, bestOption2.getValue());
+  std::vector<at::Tensor> outputs2;
+  auto duration2 = atCompl.run(
+      "blockdiagperm2dfissioned_2", {O, Idx}, outputs2, handle2, true);
+
+  // 3. Report best standalone times
+  std::cout
+      << "blockdiagperm2dfissioned_1 size I: " << I.sizes() << ", "
+      << "size W: " << W.sizes() << " ran in: "
+      << std::chrono::duration_cast<std::chrono::microseconds>(duration).count()
+      << "us\n";
+  std::cout << "blockdiagperm2dfissioned_2 size O: " << O.sizes() << ", "
+            << "size Idx: " << Idx.sizes() << " ran in: "
+            << std::chrono::duration_cast<std::chrono::microseconds>(duration2)
+                   .count()
+            << "us\n";
+
+  // 4. Run unchecked one last time, look at it in nvprof
```

Comment:
How are we supposed to differentiate checked and unchecked version under nvprof if we run the entire test?  I'd rather add an `if` on `FLAGS_use_nvprof` which we have already and run checked/unchecked accordingly.